rts/System/Sound/OpenAL/Sound.cpp

```diff
@@ -493,40 +493,58 @@ void CSound::OpenLoopbackDevice(const std::string& deviceName)
 		return;
 	}
 
+	LOG("[Sound::%s] SDL audio device(s): ", __func__);
+	for (int i = 0, n = SDL_GetNumAudioDevices(0); i < n; ++i) {
+		LOG("[Sound::%s]  * \"%d\" \"%s\"", __func__, i, SDL_GetAudioDeviceName(i, 0));
+	}
+
 	SDL_AudioSpec desiredSpec;
 	SDL_AudioSpec obtainedSpec;
 
-	desiredSpec.channels = 2;
 	desiredSpec.format = AUDIO_S16SYS;
 	desiredSpec.freq = 44100;
 	desiredSpec.padding = 0;
 	desiredSpec.samples = 4096;
 	desiredSpec.callback = RenderSDLSamples;
 	desiredSpec.userdata = this;
+	
+	/* SDL bug: can return devices with >2 channels (3D surround), even if we ask for just 2.
+ 	* (if tested) This causes the 2 "primary" channels to be moved in the 3D space compared to their "normal" state.
+ 	* (if not tested) We are afraid that this risks the 2 "primary" channels to be moved in the 3D space compared to their "normal" state, but this remains to be tested.
+ 	* For this reason, the engine rejects such devices down the road. Don't let it get that far and retry instead.
+ 	*
+ 	* Note that proper support for 3D surround sounds sounds hard, for example as of 2023 counter-strike has very weak support
+ 	* for it, apparently noticeably worse than just stereo according to players, despite being in a more relevant genre. */
+	for(int channelsDesired = 2; channelsDesired > 0; channelsDesired--) {
+		desiredSpec.channels = channelsDesired;
+
+		sdlDeviceID = 0;
+		selectedDeviceName = "";
+
+		if (!deviceName.empty()) {
+			LOG("[Sound::%s] opening configured device \"%s\"", __func__, deviceName.c_str());
+			sdlDeviceID = SDL_OpenAudioDevice(deviceName.c_str(), 0, &desiredSpec, &obtainedSpec, SDL_AUDIO_ALLOW_ANY_CHANGE & ~SDL_AUDIO_ALLOW_CHANNELS_CHANGE);
+			selectedDeviceName = deviceName;
+		}
 
-    LOG("[Sound::%s] SDL audio device(s): ", __func__);
-    for (int i = 0, n = SDL_GetNumAudioDevices(0); i < n; ++i) {
-        LOG("[Sound::%s]  * \"%d\" \"%s\"", __func__, i, SDL_GetAudioDeviceName(i, 0));
-    }
-
-	sdlDeviceID = 0;
+		if (sdlDeviceID == 0) {
+			LOG("[Sound::%s] opening default device", __func__);
+			sdlDeviceID = SDL_OpenAudioDevice(nullptr, 0, &desiredSpec, &obtainedSpec, SDL_AUDIO_ALLOW_ANY_CHANGE & ~SDL_AUDIO_ALLOW_CHANNELS_CHANGE);
+			selectedDeviceName = "default";
+		}
 
-	if (!deviceName.empty()) {
-		LOG("[Sound::%s] opening configured device \"%s\"", __func__, deviceName.c_str());
-		sdlDeviceID = SDL_OpenAudioDevice(deviceName.c_str(), 0, &desiredSpec, &obtainedSpec, SDL_AUDIO_ALLOW_ANY_CHANGE);
-		selectedDeviceName = deviceName;
-	}
+		if (sdlDeviceID == 0) {
+			LOG("[Sound::%s] failed to open SDL audio, error:  \"%s\"", __func__, SDL_GetError());
+			SDL_QuitSubSystem(SDL_INIT_AUDIO);
+			return;
+		}
 
```

Comment:
Should https://github.com/beyond-all-reason/spring/blob/5f238a9e949db6b265df3ab918afc85767a19910/rts/System/Sound/OpenAL/Sound.cpp#L530-L540 go outside of loop as the last resort solution?

Reply:
agreed